### PR TITLE
Updating the coding guidelines to recommend PoshStyle as the PowerShell coding guidelines

### PR DESCRIPTION
**Pull request by wiwei**
Our PowerShell code is currently using some random style (which is somewhat consistent within the files themselves, but are not necessarily consistent across files). This change updates our coding guidelines to recommend/reference the PoshCode style.

I've looked at some other resources both internal to Microsoft and externally visible as well (i.e. https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/strongly-encouraged-development-guidelines?view=powershell-6, and https://docs.microsoft.com/en-us/contribute/powershell/powershell-style-code) however none were as extensive as PoshCode.

### CODE REVIEW

File: Documentation/Contributing/CodingGuidelines.md [hunk 1]
```diff
@@ -559,6 +559,12 @@ DateTime.UtcNow is faster than DateTime.Now. In previous performance investigati
 
 Prefer using DateTime.UtcNow unless you actually need the localized times (a legitimate reason may be you wanting to show the current time in the user's time zone). If you are dealing with relative times (i.e. the delta between some last update and now), it's best to use DateTime.UtcNow to avoid the overhead of doing timezone conversions.
 
+## PowerShell coding conventions
+
+A subset of the MRTK codebase uses PowerShell for pipeline infrastructure and various scripts and utilities. New PowerShell code should follow the PoshCode style:
```
Comment (keveleigh) [inline Documentation/Contributing/CodingGuidelines.md]: Do we want to remove the raw URL below in favor of an in-line link?

```suggestion
A subset of the MRTK codebase uses PowerShell for pipeline infrastructure and various scripts and utilities. New PowerShell code should follow the [PoshCode style](https://poshcode.gitbooks.io/powershell-practice-and-style/).
```

Comment (keveleigh) [reply Documentation/Contributing/CodingGuidelines.md]: https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/DocumentationGuide.html#links

>Prefer links embedded in a sentence

Comment (thalbern) [reply Documentation/Contributing/CodingGuidelines.md]: Yes please!

**Comment by wiwei** [comment]
> Seems reasonable to me! Do we want to go through and update existing scripts to this style?

Short answer I think is yes, though I don't think I would trigger an immediate fixing of everything to match the style right now.

i.e. it makes sense to do it as folks make changes in the space, as they will already be testing stuff. Otherwise you run the risk of making style changes and introducing bugs. If you're already making changes and can consolidate the testing it probably makes sense

**Pull request merged**